Replace Gitter link with GH discussions. (#2837)
dotnet/source-build: README.md

# .NET Source-Build

> We now use [GitHub discussions](https://github.com/dotnet/source-build/discussions) for announcements, questions, suggestions, and sharing information with other members of the source-build community.  Please join in!

Reviewer: I prefer avoiding point in time verbiage - "We now use".  It adds negative value to readers who don't have the historical context.

```suggestion
Please use [GitHub discussions](https://github.com/dotnet/source-build/discussions) to see announcements, ask questions, make suggestions, and share information with other members of the source-build community.
```
